redfish-core/lib/assembly.hpp

```diff
@@ -278,6 +278,36 @@ void getAssemblyPresence(const std::shared_ptr<bmcweb::AsyncResp>& asyncResp,
     });
 }
 
+void getAssemblyHeath(const std::shared_ptr<bmcweb::AsyncResp>& asyncResp,
+                      const auto& serviceName, const auto& assembly,
+                      const auto& assemblyIndex)
+{
+    sdbusplus::asio::getProperty<bool>(
+        *crow::connections::systemBus, serviceName, assembly,
+        "xyz.openbmc_project.State.Decorator.OperationalStatus", "Functional",
+        [asyncResp, assemblyIndex](const boost::system::error_code& ec,
+                                   bool functional) {
+        if (ec)
+        {
+            BMCWEB_LOG_ERROR("DBUS response error {}", ec.value());
+            messages::internalError(asyncResp->res);
+            return;
+        }
+
+        nlohmann::json& assemblyArray = asyncResp->res.jsonValue["Assemblies"];
+        nlohmann::json& assemblyData = assemblyArray.at(assemblyIndex);
+
+        if (!functional)
+        {
+            assemblyData["Status"]["Health"] = "Critical";
+        }
+        else
```

Comment:
I assume we purposely don't want a default? If this is how 1020-1060 was, then yeah we want to follow this.

Reply:
Right, this incorporated https://github.com/ibm-openbmc/bmcweb/pull/628 which does not set a default value.